src/meepgeom.cpp

```diff
@@ -2599,16 +2599,16 @@ void material_grids_addgradient_point(double *v, std::complex<double> fields_a,
   }
   // no object tree -- the whole domain is the material grid
   if (!tp && is_material_grid(default_material)) {
-    vector3 pb = to_geom_box_coords(p, &tp->objects[oi]);
+    map_lattice_coordinates(p.x,p.y,p.z);
```

Comment:
I've never used this before, but I assume it's the same as above but assumes a box the size of the simulation cell (the lattice in this case)?

Maybe make sure there's no weird issues with the boundaries (when I read lattice I assume periodic boundaries, which obviously isn't always the case) just make sure the last edge in each dimension is mapped to "1" and not back to "0" (or vice-versa). 

(Ideally the two routines use the same mapping logic and the only difference is the default geometry in this case)

Reply:
The function `map_lattice_coordinates` maps elements of the `vector3` parameter `p` into the range [0,1]:

https://github.com/NanoComp/meep/blob/8e2b0a050fd7e2d6d39019f76cdd3abcb73888c5/src/meepgeom.cpp#L374-L381

I think you were the one that set up the mapping this way. As long as `p` is defined to be within `geometry_lattice`, it should map points along the boundaries correctly.